src/server.rs

```diff
@@ -0,0 +1,61 @@
+use crate::verkle::{get_rlp, decode_block, verification};
+use crate::dot::to_dot;
+use actix_web::{web::{self, Json}, App, HttpServer};
+use crate::types::{VerkleReq, VerkleResp};
+use crate::Config;
+use tokio::process::Command;
+
+async fn get_block_info(data: Json<VerkleReq>) -> Result<Json<VerkleResp>, crate::error::Error> {
+    let block_number = data.into_inner().block_number;
+    let block_rlp = get_rlp(block_number).await?;
+    let block = decode_block(block_rlp.clone())?;
+
+    if block_number < 2 {
+        return Ok(Json(VerkleResp { image: "incorrect block number".to_owned() }));
+    }
+
+    let previous_block_rlp = get_rlp(block_number - 1).await?;
+    let previous_block = decode_block(previous_block_rlp)?;
+
+    let parent_root = hex::encode(previous_block.header.storage_root);
```

Comment:
Why do you encode it? As I can see, you do not use it as a String and will decode it back in `algorithm::verification` anyway